plugins/woocommerce/src/StoreApi/Routes/V1/Checkout.php

```diff
@@ -391,9 +391,23 @@ private function create_or_update_draft_order( \WP_REST_Request $request ) {
 		 * If POSTing to the checkout (attempting to pay), set the timeout to 60 mins (using the woocommerce_hold_stock_minutes option).
 		 */
 		try {
+
 			$reserve_stock = new ReserveStock();
-			$duration      = $request->get_method() === 'POST' ? (int) get_option( 'woocommerce_hold_stock_minutes', 60 ) : 10;
+
+			/**
+			 * Filters the hold stock duration in minutes for draft orders on checkout entry.
+			 *
+			 * This hook filters the duration in minutes that stock is held for draft orders on checkout entry, it allows a third party to amend the duration to increase/reduce the time the stock is held for draft orders on checkout entry.
+			 *
+			 * @since 8.8.0
+			 *
+			 * @param integer $minutes Minutes to hold stock for draft orders on checkout entry.
+			 */
+			$draft_order_hold_stock_minutes = (int) apply_filters( 'woocommerce_draft_order_hold_stock_minutes', 10 );
```

Comment:
Maybe we should check that the result is: not empty, is an integer (or can be cast to one) and is not negative, if any of these are true it should default back to 10. Do you agree?

Reply:
@opr I think casting to a integer and a check for negative is find, however the !empty would stop being able to filter to 0, which for our particular scenario (to be stop stock reservation) is why we need this filter.

Reply:
Yep, you're right 👍🏼

Reply:
The int casting is already there, so shall I just add a negative check here?